lisa/analysis/frequency.py

```diff
@@ -338,15 +338,19 @@ def get_average_cpu_frequency(self, cpu):
         return series_mean(freq)
 
     @TraceAnalysisBase.cache
-    @requires_events('clock_set_rate', 'clock_enable', 'clock_disable')
+    @requires_events('clk_set_rate', 'clk_enable', 'clk_disable')
     def df_peripheral_clock_effective_rate(self, clk_name):
-        rate_df = self.trace.df_event('clock_set_rate')
-        enable_df = self.trace.df_event('clock_enable')
-        disable_df = self.trace.df_event('clock_disable')
+        rate_df = self.trace.df_event('clk_set_rate')
+        enable_df = self.trace.df_event('clk_enable')
+        disable_df = self.trace.df_event('clk_disable')
 
-        freq = rate_df[rate_df.clk_name == clk_name]
-        enables = enable_df[enable_df.clk_name == clk_name]
-        disables = disable_df[disable_df.clk_name == clk_name]
+        # Add 'state' for enable and disable events
+        enable_df['state'] = 1
+        disable_df['state'] = 0
+
+        freq = rate_df[rate_df.name == clk_name]
```

Comment:
column access must be done with `[]`, e.g. `rate_df]'name']`. Pandas has a quite flaky API so doing that prevents a bunch of issues and make some others more clear.